lte/gateway/c/session_manager/ShardTracker.cpp

```diff
@@ -13,45 +13,58 @@
 
 #include <string>
 #include <vector>
+#include <set>
 
 #include "ShardTracker.h"
 
 namespace magma {
 
-int ShardTracker::add_ue() {
-  /*
-  * If there are no shards, add an entry for the shard
-  * representing a single UE at the new shard
-  */
-  if (ue_count_per_shard_.size() == 0) {
-    ue_count_per_shard_.push_back(1);
-    return 0;
+int ShardTracker::add_ue(std::string imsi) {
+  /**
+   * If there are no shards, add a single entry set
+   * to shards list containing the IMSI
+   * */
+  if (shards_.empty()) {
+    std::set<std::string> new_shard = {imsi};
+    shards_.push_back(new_shard);
```

Comment:
I still don't understand well this robject but looks like it may use singleton pattern
https://www.tutorialspoint.com/Explain-Cplusplus-Singleton-design-pattern
https://stackoverflow.com/questions/1008019/c-singleton-design-pattern

Is ShardTracker a set of buckets where to place IMSIS? Note a design document with pictures  would help a lot to visual

Reply:
So the object is a vector of sets, where each set holds the IMSIs associated with the shard id.

Reply:
I will also add a small diagram to the design doc to make this clear.